testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/util/IcebergTestUtils.java

```diff
@@ -25,13 +21,11 @@ private IcebergTestUtils() {}
 
     public static String getTableLocation(String tableName)
     {
-        Pattern locationPattern = Pattern.compile(".*location = 'hdfs://hadoop-master:9000(.*?)'.*", Pattern.DOTALL);
-        Matcher m = locationPattern.matcher((String) onTrino().executeQuery("SHOW CREATE TABLE " + tableName).getOnlyValue());
-        if (m.find()) {
-            String location = m.group(1);
-            verify(!m.find(), "Unexpected second match");
-            return location;
-        }
-        throw new IllegalStateException("Location not found in SHOW CREATE TABLE result");
+        return (String) onTrino().executeQuery("SELECT DISTINCT regexp_replace(\"$path\", '/[^/]*/[^/]*$', '') FROM " + tableName).getOnlyValue();
+    }
+
+    public static String stripNamenodeURI(String location)
+    {
+        return location.replaceFirst("hdfs://hadoop-master:9000", "");
```

Comment:
Input from @hashhar  : What about using `URI.getPath(...)`

See that it is used already in hive product tests

https://github.com/trinodb/trino/blob/cac8bfecdbd9e4478c3cda5a1a0f6014e393b5e2/testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/util/TableLocationUtils.java#L60-L66

Reply:
Yes, make sense. Used `URI.create(location).getPath();`